compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -532,6 +534,22 @@ impl<'ll, 'tcx> IntrinsicCallBuilderMethods<'tcx> for Builder<'_, 'll, 'tcx> {
                 return Ok(());
             }
 
+            sym::dynamic_shared_memory => {
+                let global = self.declare_global_in_addrspace(
+                    "dynamic_shared_memory",
```

Comment:
Is there a special magic meaning to this hard-coded name for the global?

When linking together Rust code and code in other languages, is this the name everyone must use consistently to ensure things behave correctly?

Reply:
The name has no meaning.
I didn’t try, but I’m rather confident to claim that *all external addrspace(3) globals* alias. There is no way to get two different dynamic shared memory pointers.
Using the name here and updating the global’s alignment is to generate readable and “clean” LLVM IR.